### DISCUSSION TITLE
How to Find the Intersection Between Two `pystac` Item Collections?

### DISCUSSION THREAD

**Discussion by elliesch** [microsoft/PlanetaryComputer#289] [Q&A] [answered]
I have defined a coordinate bounding box that I'm using to query two Planetary Computer catalogs, that I then store in two seperate `pystac` Feature Collections.

```python
# first search
search_one = catalog.search(collections=["example-cat1"], 
                            bbox=latlon_bounds, 
                            datetime=time_range)
items_one = search_one.get_all_items()

# second search
search_two = catalog.search(collections=["example-cat2"], 
                            bbox=latlon_bounds, 
                            datetime=time_range)
items_two = search_two.get_all_items()
```

How do I find the intersection of `items_one` and `items_two` on their lat/lon coordinates using `pystac` (or something else)? I'm interested in the inner join between both datasets. The end goal is to create something like a `stackstac` mosaic which would provide an `xarray` including the intersection of both, with each catalog as separate bands within a unified dataset.

Thank you in advance for any help with this!

**Comment by TomAugspurger** [comment, marked as answer]
Do you want a single geometry as your output? Or a collection of geometries (where each on is like the intersection between two input items)?

I think the second one would be a spatial join, which is implemented in geopandas: https://geopandas.org/en/stable/gallery/spatial_joins.html. We have a small example of geopandas + STAC at https://planetarycomputer.microsoft.com/docs/quickstarts/reading-stac/#Analyzing-STAC-Metadata.

**Reply by elliesch** [reply]
Thank you so much for your help @TomAugspurger! I'm new to geospatial data and didn't know where to look. The `geopandas` spatial join is exactly what I was looking for. Thank you for the useful tutorials!
